src/ripple/core/impl/Config.cpp

```diff
@@ -162,7 +162,12 @@ void Config::setup (std::string const& strConf, bool bQuiet)
     // config directory and that with "db" as the data
     // directory.
 
-    QUIET = bQuiet;
+    QUIET = bQuiet || bSilent;
+    SILENT = bSilent;
+    RUN_STANDALONE = bStandalone;
+
+    if (RUN_STANDALONE)
+        LEDGER_HISTORY = 0;
```

Comment:
This is broken: we call `load` after this statement (around 233), which then calls `loadFromString` which sets `LEDGER_HISTORY` based on the `[ledger_history]` entry from the config file, overwriting this value.